compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -2545,9 +2545,15 @@ public Set<BVarSymbol> getConfigVarSymbolsIncludingImportedModules(BPackageSymbo
 
     private void populateConfigurableVars(BPackageSymbol pkgSymbol, Set<BVarSymbol> configVars) {
         for (Scope.ScopeEntry entry : pkgSymbol.scope.entries.values()) {
-            BSymbol symbol = entry.symbol.tag == SymTag.TYPE_DEF ? entry.symbol.type.tsymbol : entry.symbol;
-            if (symbol != null && symbol.tag == SymTag.VARIABLE && Symbols.isFlagOn(symbol.flags, Flags.CONFIGURABLE)) {
-                configVars.add((BVarSymbol) symbol);
+            BSymbol symbol = entry.symbol;
+            if (symbol != null) {
```

Comment:
Was it a symbol from an improperly resolved dependency that was null here?

Reply:
Yes. It was the `os` symbol from the `file` import

Reply:
So this would just delay the failure, right? Shall we add a comment on https://github.com/ballerina-platform/ballerina-lang/issues/35385 to undo this change once that is properly fixed? If not, we may end up with quite a few unnecessary checks, right?

Reply:
No right. This fix should be there. It's wrong anyway to access a null. ie; assuming the symbol would not be null is wrong IMO. I understand it's irrespective of the actual bug

Reply:
Given that we are looping through the entries defined in the scope, IMO, we can assume them to be not null. For example, we don't have the null check even in `org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolResolver#lookupSymbol` which is called frequently. I don't think we would define it in the scope if the symbol is null.